app/components/newsletter/newsletter-header-overview.js

```diff
@@ -106,73 +72,151 @@ export default class NewsletterHeaderOverviewComponent extends Component {
 
     await reloadedMeeting.save();
     this.isLoading = false;
-    location.reload();
   }
 
   @action
-  async sendCampaign() {
+  async publishToMail() {
     this.isLoading = true;
-    const agenda = this.args.agenda;
-    const meeting = this.args.meeting;
-    const mailCampaign = await meeting.mailCampaign;
+    const mailCampaign = await this.getMailCampaign();
 
-    if (!mailCampaign || !mailCampaign.id || mailCampaign.isSent) {
-      return;
+    if (mailCampaign) {
+      if (await this.validateMailCampaign(mailCampaign)) {
+        await this.publishToMailAndSaveCampaign(mailCampaign);
+      }
+    }
+
+    this.isLoading = false;
+    this.toggleVerifyingPublishMail();
+  }
+
+  @action
+  async publishToBelga() {
+    this.isLoading = true;
+    await this.publishNewsletterToBelga();
+    this.isLoading = false;
+    this.toggleVerifyingPublishBelga();
+  }
+
+  @action
+  async publishToAll() {
+    this.isLoading = true;
+
+    const mailCampaign = await this.getMailCampaign();
+    if (mailCampaign) {
+      if (await this.validateMailCampaign(mailCampaign)) {
+        await this.publishToMailAndSaveCampaign(mailCampaign);
+      }
+    }
+    await this.publishNewsletterToBelga;
+
+    this.isLoading = false;
+    this.toggleVerifyingPublishAll();
+  }
+
+  @action
+  async showNewsletter() {
+    this.loadingNewsletter = true;
+
+    const mailCampaign = await this.getMailCampaign();
```

Comment:
voorbeeldje hier, vanuit gaande van het ticket " Het aanmaken van nieuwsbrief zit verwerkt in elke actie waar dit voor nodig is."
met een tracked `this.mailCampaign" zou je dan eerder kunnen doen:
```
if (!this.mailCampaign) {
 await this.createCampaign();
}
```
Dan ben je ook zeker dat je vanaf dan altijd een `this.mailCampaign` kan aanspreken (tenzij errors gegooid worden bij creatie natuurlijk)

Reply:
en dan is de code transparanter